kork/kork-artifacts/src/main/java/com/netflix/spinnaker/kork/artifacts/artifactstore/entities/ManifestMapStorageHandler.java

```diff
@@ -0,0 +1,139 @@
+/*
+ * Copyright 2025 Apple Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.kork.artifacts.artifactstore.entities;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.netflix.spinnaker.kork.artifacts.ArtifactTypeDecorator;
+import com.netflix.spinnaker.kork.artifacts.ArtifactTypes;
+import com.netflix.spinnaker.kork.artifacts.artifactstore.ArtifactStore;
+import com.netflix.spinnaker.kork.artifacts.artifactstore.filters.ApplicationFilter;
+import com.netflix.spinnaker.kork.artifacts.artifactstore.filters.ApplicationStorageFilter;
+import com.netflix.spinnaker.kork.artifacts.model.Artifact;
+import com.netflix.spinnaker.security.AuthenticatedRequest;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * ManifestStorageHandler is used to convert manifest values into artifacts. This is done by
+ * matching against a key within the execution context. Upon matching, we will convert the manifest
+ * into an artifact and write the artifact in its place.
+ */
+public class ManifestMapStorageHandler implements ArtifactStorageHandler {
+  private List<String> keys = List.of("manifests", "outputs.manifests");
+  private final ApplicationFilter exclude;
+
+  public ManifestMapStorageHandler(Map<String, List<ApplicationStorageFilter>> exclude) {
+    this.exclude =
+        new ApplicationFilter(exclude.get(ArtifactTypes.EMBEDDED_MAP_BASE64.getMimeType()));
+  }
+
+  /** Manifest storage can only be handled on a list of maps. */
```

Comment:
I could use some more comments here.  From looking at the code, this returns false unless `v` is a map, which at first glance means the code handles maps, but not "a list of maps" like the comment says.